### ISSUE TITLE
Does not detect sentences ending with ! or ?

### ISSUE THREAD

**Issue by JoshuaKGoldberg** [JoshuaKGoldberg/sentences-per-line#22]
Right now only `.` is detected:

https://github.com/JoshuaKGoldberg/sentences-per-line/blob/bea2b68340989f637bda693fea548bd874a029be/index.js#L88

So this test case does not work:

```md
Hello world! Another sentence!
```

How unfortunate.

**Comment by AzineZ** [comment]
Hi! I'm a student in the CodePath AI301 program and I'd like to work on this issue. I'll start by reproducing the problem locally this week. If you have any pointers on where to start in the codebase, I would really appreciate it!

**Comment by JoshuaKGoldberg** [comment]
Cool, welcome! https://github.com/JoshuaKGoldberg/sentences-per-line/blob/51a853bf26f1efdf0eaa6a0913f0a7577bcba737/packages/sentences-per-line/src/getIndexBeforeSecondSentence.ts is a good starting point: I'd read through the comments/logic in that file and see where else `getIndexBeforeSecondSentence` is used in code.

**Comment by AzineZ** [comment]
Thank you! I have some questions specifically about the dot itself:
1. Right now, ". World" will actually return the index instead of undefined. This means that the index of the second sentence is found. But in the test, there is no first sentence, only the dot. Is this an expected behavior?
2. "Hello.  World" (which two whitespaces) returns undefined. Is this also an expected behavior?

**Comment by JoshuaKGoldberg** [comment]
I'm low bandwidth this month and I don't understand either of your questions. Maybe I'm just misreading them, or it's a language/internet-communications barrier. Could you rephrase?

**Comment by AzineZ** [comment]
So in the getIndexBeforeSecondSentence function, it returns the index of where the first sentence ends, otherwise it returns undefined. So for a test like "Hello. World", the index returned is 6, so right where the dot is. And the way the dot is implemented right now when I forked the repo is that if it sees a dot, then a whitespace, then a new letter that is capitalized, it will return the index of this dot, indicating that a sentence just ended.

So in the two tests I brought up, the first test returns 1, which is the first index, where the dot is. And in the second test, since there are two whitespaces after "Hello.", it doesn't mark the end of the first sentence, so what returned is "undefined".

I want to ask if the dot behaves as expected since I plan on implementing the behaviors the ! and ? similar to the dot.

**Comment by JoshuaKGoldberg** [comment]
Please use your human intuition on this.

Not trying to be gruff/rude, just I'm dealing with a lot of AI spam and it's hard to tell who is a genuine human vs bot asking me to clarify prompts.

**Comment by AzineZ** [comment]
Yeah, that's absolutely understandable. I just want to give you some updates the last time we talked. I implemented the ! and ? locally and they should behave similar to the dot. I will run it through some tests and have a PR ready some time next week.
